syntax = "proto3";

package syft.grid.messages;

import "proto/core/common/common_object.proto";
import "proto/core/io/address.proto";
import "proto/lib/python/dict.proto";


// CREATE Role
message CreateRoleMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  string name =  3;
  bool can_triage_requests = 4;
  bool can_edit_settings = 5;
  bool can_create_users = 6;
  bool can_create_groups = 7;
  bool can_edit_roles = 8;
  bool can_manage_infrastructure = 9;
  bool can_upload_data = 10;
  syft.core.io.Address reply_to = 11;
}


// GET Role
message GetRoleMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  int32 role_id = 3;
  syft.core.io.Address reply_to = 4;
}


message GetRoleResponse {
  syft.core.common.UID msg_id = 1;
  syft.lib.python.Dict content = 2;
  syft.core.io.Address address = 3;
}

// GET ALL Roles
message GetRolesMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  syft.core.io.Address reply_to = 4;
}

message GetRolesResponse {
  syft.core.common.UID msg_id = 1;
  repeated syft.lib.python.Dict content = 2;
  syft.core.io.Address address = 3;
}


// UPDATE Role
message UpdateRoleMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  string name =  3;
  bool can_triage_requests = 4;
  bool can_edit_settings = 5;
  bool can_create_users = 6;
  bool can_create_groups = 7;
  bool can_edit_roles = 8;
  bool can_manage_infrastructure = 9;
  bool can_upload_data = 10;
  int32 role_id = 11;
  syft.core.io.Address reply_to = 12;
}

// Delete Role
message DeleteRoleMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  int32 role_id = 3;
  syft.core.io.Address reply_to = 4;
}

